KIPs/kip-dreaft_sdk_common_architecture.md

```diff
@@ -0,0 +1,1623 @@
+| ----------- | ----------- |
+| sign(txHash: String, chainId: int, role: int): List&#60;SignatureData&#62; | Signs the transaction using key(s) defined in role. `sign` is defined as an "abstract method" in AbstractKeyring, and must be implemented in all keyring classes that extend `AbstractKeyring`. |
+| sign(txHash: String, chainId: int, role: int, index: int): SignatureData | Signs the transaction using a key defined in role. `sign` is defined as an "abstract method" in AbstractKeyring, and must be implemented in all keyring classes that extend `AbstractKeyring`. |
+| signMessage(message: String, role: int): MessageSigned | Signs the message using key(s) defined in role. `signMessage` is defined as an "abstract method" in AbstractKeyring, and must be implemented in all keyring classes that extend `AbstractKeyring`. |
+| signMessage(message: String, role: int, index: int): MessageSigned | Signs the message using a key defined in role. `signMessage` is defined as an "abstract method" in AbstractKeyring, and must be implemented in all keyring classes that extend `AbstractKeyring`. |
+| encrypt(password: String, options: Object): Object | Encrypts a Keyring instance with keystore v4 format. `encrypt` is defined as an "abstract method" in AbstractKeyring, and must be implemented in all keyring classes that extend `AbstractKeyring`. |
+| copy(): AbstractKeyring | Returns a copied Keyring instance. `copy` is defined as an "abstract method" in AbstractKeyring, and must be implemented in all keyring classes that extend `AbstractKeyring`. |
+| sign(txHash: String, chainId: String, role: int): List&#60;SignatureData&#62; | Signs the transaction using key(s) defined in role. Converts the type of chainId inputted as String into int and calls the sign function implemented in keyring. |
+| sign(txHash: String, chainId: String, role: int, index: int): SignatureData | Signs the transaction using a key defined in role. Converts the type of chainId inputted as String into int and calls the sign function implemented in keyring. |
+| getKlaytnWalletKey(): String | Returns KlaytnWalletKey string. getKlaytnWalletKey is implemented to throw exception by default in `AbstractKeyring`. In the case of `SingleKeyring` that can use KlaytnWalletKey format, getKlaytnWalletKey function must be overridden. |
+| encrypt(password: String): Object | Encrypts a Keyring instance with keystore v4 format. |
+| encryptV3(password: String): Object | Encrypts a Keyring instance with keystore v3 format. encryptV3 is implemented to throw exception by default in `AbstractKeyring`. In the case of `SingleKeyring` that can encrypt to keystore v3, encryptV3 function must be overridden. |
+| encryptV3(password: String, options: Object): Object | Encrypts a Keyring instance with keystore v3 format. encryptV3 is implemented to throw exception by default in `AbstractKeyring`. In the case of `SingleKeyring` that can encrypt to keystore v3, encryptV3 function must be overridden. |
```

Comment:
```suggestion
| encryptV3(password: String, options: Object): Object | Encrypts a Keyring instance with keystore v3 format. |
```

Reply:
Need to specify the difference between `encryptV3(password: String): Object` and `encryptV3(password: String, options: Object): Object`.